src/arkode/arkode_mristep.c

```diff
@@ -1413,239 +1520,1440 @@ int mriStep_FullRHS(ARKodeMem ark_mem, sunrealtype t, N_Vector y, N_Vector f,
+  N_Vector ytemp;                     /* temporary vector           */
+  SUNAdaptController_Type adapt_type; /* timestep adaptivity type   */
+  sunbooleantype embedding;           /* flag indicating embedding  */
+  sunbooleantype solution;            /*   or solution stages       */
+  sunbooleantype impl_corr;           /* is slow correct. implicit? */
+  sunrealtype cstage;                 /* current stage abscissa     */
+  sunbooleantype need_inner_dsm;
+  int nvec, max_stages;
+  const sunrealtype tol = SUN_RCONST(100.0) * SUN_UNIT_ROUNDOFF;
+
+  /* access the MRIStep mem structure */
+  retval = mriStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* initialize algebraic solver convergence flag to success;
+     error estimate to zero */
+  *nflagPtr = ARK_SUCCESS;
+  *dsmPtr   = ZERO;
+
+  /* set N_Vector shortcuts */
+  ytilde = ark_mem->tempv4;
+  ytemp  = ark_mem->tempv2;
+
+  /* if MRI adaptivity is enabled: reset fast accumulated error,
+     and send appropriate control parameter to the fast integrator */
+  adapt_type     = SUNAdaptController_GetType(ark_mem->hadapt_mem->hcontroller);
+  need_inner_dsm = SUNFALSE;
+  if (adapt_type == SUN_ADAPTCONTROLLER_MRI_H_TOL)
+  {
+    need_inner_dsm      = SUNTRUE;
+    step_mem->inner_dsm = ZERO;
+    retval = mriStepInnerStepper_ResetAccumulatedError(step_mem->stepper);
+    if (retval != ARK_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_INNERSTEP_FAIL, __LINE__, __func__, __FILE__,
+                      "Unable to reset the inner stepper error estimate");
+      return (ARK_INNERSTEP_FAIL);
+    }
+    retval = mriStepInnerStepper_SetRTol(step_mem->stepper,
+                                         step_mem->inner_rtol_factor *
+                                           ark_mem->reltol);
+    if (retval != ARK_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_INNERSTEP_FAIL, __LINE__, __func__, __FILE__,
+                      "Unable to set the inner stepper tolerance");
+      return (ARK_INNERSTEP_FAIL);
+    }
+  }
+
+  /* for adaptive computations, reset the inner integrator to the beginning of this step */
+  if (!ark_mem->fixedstep)
+  {
+    retval = mriStepInnerStepper_Reset(step_mem->stepper, ark_mem->tn,
+                                       ark_mem->yn);
+    if (retval != ARK_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_INNERSTEP_FAIL, __LINE__, __func__, __FILE__,
+                      "Unable to reset the inner stepper");
+      return (ARK_INNERSTEP_FAIL);
+    }
+  }
+
+  /* Evaluate the slow RHS functions if needed. NOTE: we decide between calling the
+     full RHS function (if ark_mem->fn is non-NULL) versus just updating the stored
+     values of Fse[0] and Fsi[0].  In either case, we use ARK_FULLRHS_START mode
+     because MRISR methods do not evaluate the RHS functions at the end of the
+     time step (so nothing can be leveraged). */
```

Comment:
I think this will need the same nested MRI logic for calling `UpdateF0` instead of `FullRHS` suggested above.